draft-ietf-jose-hpke-encrypt.md

```diff
@@ -106,167 +109,222 @@ informative:
+| HPKE-5 | DHKEM(X448, HKDF-SHA512)   | HKDF-SHA512 | AES-256-GCM      |
+| HPKE-6 | DHKEM(X448, HKDF-SHA512)   | HKDF-SHA512 | ChaCha20Poly1305 |
+| HPKE-7 | DHKEM(P-256, HKDF-SHA256)  | HKDF-SHA256 | AES-256-GCM      |
++--------+----------------------------+-------------+------------------+
+~~~
+{: #ciphersuite-int-algs title="Algorithms using HPKE for Integrated Encryption"}
+
+The HPKE KEM, KDF, and AEAD values are chosen from the IANA HPKE registry {{IANA.HPKE}}.
+
+## JWE Compact Serialization Example {#compact-example}
+
+Below is an example of a JWE using the Compact Serialization and Integrated Encryption with HPKE:
 
 ~~~
-{::include-fold examples/compact_example.txt}
+eyJhbGciOiAiSFBLRS0wIiwgImVuYyI6ICJpbnQiLCAia2lkIjogIkc1Tl9fQ3FNdl9rSkdpZUdTRnVBdWd2bDBqclFKQ1ozeUt3Vks2c1VNNG8ifQ.BIh6I40uiBbK8-UK7nHdo3ISEfgwJ_MF3zWjQzLt00GhFF2-1VgWKHSYLXdeVeRV7AinyocYiCYmISvW0yqiDmc..Ov-llz6VUyiw8nZL0OPGLGZckLTm5UcTZFg.
 ~~~
 
-The keys used for this example are in {{keys-used}}.
+The key used for this example is in {{int-key}}.
 
 # Key Encryption
 
 When using the JWE JSON Serialization,
-recipients using JOSE-HPKE can be added alongside other recipients
+recipients using Key Encryption with HPKE can be added alongside other recipients
 (e.g., those using `ECDH-ES+A128KW` or `RSA-OAEP-256`),
-since HPKE is used to encrypt the Content Encryption Key,
-which is then processed as specified in JWE.
-
-The encoding of the protected header remains consistent with existing JWE rules.
+since HPKE is used to encrypt the Content Encryption Key (CEK).
 
-In HPKE JWE Key Encryption:
+When using Key Encryption with HPKE:
 
-- The Key Management Mode is Key Encryption.
-- When all recipients use the same HPKE algorithm to secure the Content Encryption Key, the JWE Protected Header SHOULD contain "alg".
+- When all recipients use the same JWE algorithm to secure the Content Encryption Key, the JWE Protected Header SHOULD contain "alg".
 Otherwise, the JWE Protected Header (and JWE Shared Unprotected Header) MUST NOT contain "alg".
-- JOSE Header parameter "alg" MUST be a JOSE-HPKE algorithm.
-- JOSE Header parameter "psk_id" MAY be present.
-- JOSE Header parameter "ek" MUST be present and contain the base64url-encoded HPKE encapsulated key.
-- Recipient JWE Encrypted Key MUST be the ciphertext from HPKE Encryption.
-- The HPKE info parameter contains the encoding of the Recipient_structure, which is described in {{recipient_structure}}.
-- The HPKE AAD parameter defaults to the empty string; externally provided information MAY be used instead.
+- The "alg" header parameter MUST be a JWE algorithm using Key Encryption.
+- The header parameter "psk_id" MAY be present.
+- The header parameter "ek" MUST be present and contain the base64url-encoded HPKE encapsulated secret.
+- The HPKE aad parameter MUST be set to the "Additional Authenticated Data encryption parameter" value specified in Step 15 of {{encryption}}.
```

Comment:
The HPKE aad parameter is empty for Key Encryption. Using Additional Authenticated Data encryption parameter would cause extreme implementation difficulties.

Reply:
I would say:  
The HPKE aad parameter MUST be set to a empty string.

Reply:
@selfissued this section is for HPKE Key Encryption, so no, we don't use HPKE to protect the JWE AAD, as that is done by JWE's "enc" / AEAD as per usual. In HPKE Key Encryption HPKE only protects the CEK and the cross-layer Recipient_structure.

Reply:
AADEP => Additional Authenticated Data encryption parameter

Since encryption procedure is supposed to be in order, that value is only supposed to appear in bulk encryption (step 16 or 17), not in key encryption (step 4).